providers/env-var/src/main/java/dev/openfeature/contrib/providers/envvar/EnvironmentKeyTransformer.java

```diff
@@ -0,0 +1,109 @@
+ *     .hyphenCaseToScreamingSnake();
+ *
+ * FeatureProvider provider = new EnvVarProvider(transformer);
+ * }
+ * </pre>
+ * 2. chained/composed transformations:
+ * <pre>
+ * {@code
+ * // Definition of the EnvVarProvider:
+ * EnvironmentKeyTransformer transformer = EnvironmentKeyTransformer
+ *     .toLowerCaseTransformer()
+ *     .andThen(EnvironmentKeyTransformer.replaceUnderscoreWithDotTransformer());
+ *
+ * FeatureProvider provider = new EnvVarProvider(transformer);
+ * }
+ * </pre>
+ * 3. freely defined transformation function:
+ * <pre>
+ * {@code
+ *
+ * // Definition of the EnvVarProvider:
+ * EnvironmentKeyTransformer transformer = new EnvironmentKeyTransformer(key -> "constant");
+ *
+ * FeatureProvider provider = new EnvVarProvider(keyTransformer);
+ * }
+ * </pre>
+ */
+@RequiredArgsConstructor
+public class EnvironmentKeyTransformer {
+
+    private static final UnaryOperator<String> TO_LOWER_CASE = StringUtils::lowerCase;
+    private static final UnaryOperator<String> TO_UPPER_CASE = StringUtils::upperCase;
+    private static final UnaryOperator<String> TO_CAMEL_CASE = s -> CaseUtils.toCamelCase(s, false, '_');
+    private static final UnaryOperator<String> REPLACE_DOT_WITH_UNDERSCORE = s -> StringUtils.replaceChars(s, ".", "_");
+    private static final UnaryOperator<String> REPLACE_UNDERSCORE_WITH_DOT = s -> StringUtils.replaceChars(s, "_", ".");
+    private static final UnaryOperator<String> REPLACE_HYPHEN_WITH_UNDERSCORE =
+        s -> StringUtils.replaceChars(s, "-", "_");
+
+    private final Function<String, String> transformation;
+
+    public String transformKey(String key) {
+        return transformation.apply(key);
+    }
+
+    public EnvironmentKeyTransformer andThen(EnvironmentKeyTransformer another) {
+        return new EnvironmentKeyTransformer(transformation.andThen(another::transformKey));
+    }
+
+    public static EnvironmentKeyTransformer toLowerCaseTransformer() {
+        return new EnvironmentKeyTransformer(TO_LOWER_CASE);
+    }
+
+    public static EnvironmentKeyTransformer toUpperCaseTransformer() {
+        return new EnvironmentKeyTransformer(TO_UPPER_CASE);
+    }
+
+    public static EnvironmentKeyTransformer toCamelCaseTransformer() {
+        return new EnvironmentKeyTransformer(TO_CAMEL_CASE);
+    }
+
+    public static EnvironmentKeyTransformer replaceUnderscoreWithDotTransformer() {
+        return new EnvironmentKeyTransformer(REPLACE_UNDERSCORE_WITH_DOT);
+    }
+
+    public static EnvironmentKeyTransformer replaceDotWithUnderscoreTransformer() {
+        return new EnvironmentKeyTransformer(REPLACE_DOT_WITH_UNDERSCORE);
+    }
+
+    public static EnvironmentKeyTransformer hyphenCaseToScreamingSnake() {
+        return new EnvironmentKeyTransformer(REPLACE_HYPHEN_WITH_UNDERSCORE)
+            .andThen(EnvironmentKeyTransformer.toUpperCaseTransformer());
+    }
+
+    public static EnvironmentKeyTransformer doNothing() {
```

Comment:
NoOp is the convention I see it in open feature SDK implementation for this kind of scenarios. I'm okay with `doNothing` as well. @Kavindu-Dodan @toddbaert any thoughts on this?

Reply:
Good point. Thinking deeper here, this is an `identity function similar to Function.identity [1] rather than a `NoOp`(performing nothing vs returning input itself)

Maybe we can rename(potentially breaking change)/deprecate & introduce a new function () for a later release :)

[1] - https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html#identity--

Reply:
yes.. I prefer to deprecate the `doNothing` and use the `Function.identity`. I will find sometime during the week and open a PR.

Reply:
I explicitly wanted to provide a `doNothing`/`noOp` transformer to let the consumer of the api not to bother, if they don't need to. Having said that, using `Function.identity` instead of `s -> s` is certainly an improvement. 

So, I would like to see a dedicated `doNothing` or `noOp`/`NO_OP` transformer and the replacement of `s -> s` with `Function.identity`. What are your thoughts?

Reply:
@lopitz I feel this as a good improvement :) feel free to open a PR for this